### ISSUE TITLE
Is cFAC Parallelizable?

### ISSUE THREAD

**Issue by 199718Akash** [fnevgeny/cfac#15]
I am trying to run cFAC on a supercomputer with one node containing 48 processors. But the code exits with the following errors: 

File type is neither DB_EN nor DB_ENF
No levels found in the DB file Ar.bin!
Energy level table has not been built in memory.
File type is neither DB_EN nor DB_ENF
No levels found in the DB file Ar.bin!
Energy level table has not been built in memory.
No levels found in the DB file Ar.bin!
Energy level table has not been built in memory.
Error parsing file Ar.bin!
Error parsing file Ar.bin!
Error parsing file Ar.bin!
Error parsing file Ar.bin!
Energy level table has not been built in memory.
Error parsing file Ar.bin!
Energy level table has not been built in memory.
Error parsing file Ar.bin!
Energy level table has not been built in memory.

However, If I use only 1 processor, there seems to be no problem at all. 

A resolution to this error will help a lot. 
Thanks in advance!

**Comment by fnevgeny** [comment]
No, cFAC isn't parallelized. For this reason, no matter how many CPUs are present in your computer, it will only use a single one. So I don't understand what you're really doing to get these weird errors...

**Comment by 199718Akash** [comment]
Thanks for the reply. I guess these errors pop up as result of assigning the code to multiple processors. 

Do you have any idea of the computation time, if configurations in more than 3 complexes are allowed to interact together for energy level calculations. I am using the following configuration for the neutral Ar atom: (Closed('1s') is specified in the beginning)
Config('2x8 3x5 4x1 5x1 6x1', group = '4x1 5x1 6x1') 

I gather that the convergence will be slow in this case using CI treatment but if I can get some idea of the time taken, it will really help a lot.

The call to optimize radial is done using the ground state configuration of neutral Ar atom.

**Comment by fnevgeny** [comment]
>  I guess these errors pop up as result of assigning the code to multiple processors.

Do you mean you run multiple instances of sfac in parallel writing to the same out files?? There would be a mess, of course. There would be a mess with any other code, parallelized or not.

> Do you have any idea of the computation time

Nope.

> The call to optimize radial is done using the ground state configuration of neutral Ar atom.

I assume you realize that the results would be of a fairly inferior accuracy, don't you? FAC isn't supposed to be used for multi-electron neutral atoms. Twice or trice-ionized ion is a reasonable minimum.

**Comment by 199718Akash** [comment]
Yes apparently, I was running multiple parallel instances of sfac while writing to the same output files. I did that assuming cFAC was parallelized. I get that now, it would result in a blunder even with a  parallelized code. 

I do realize that the results would be least accurate for neutral atoms. I chose neutral Ar atom just as a starting point to get an idea of the computation time. 

Either ways thanks for the clarifications !
